corehq/apps/dropbox/tasks.py

```diff
@@ -10,43 +13,68 @@
 from corehq.util.translation import localize
 from corehq.util.log import send_HTML_email
 
+CHUNK_SIZE = 5000000  # Read file in 5 MB chunks
+
 
 @task
 def upload(dropbox_helper_id, access_token, size, max_retries):
     from .models import DropboxUploadHelper
     helper = DropboxUploadHelper.objects.get(id=dropbox_helper_id)
-    client = DropboxClient(access_token)
-    retries = 0
+    dbx = Dropbox(access_token)
 
     try:
         with open(helper.src, 'rb') as f:
-            uploader = client.get_chunked_uploader(f, size)
-            while uploader.offset < size:
-                helper.progress = uploader.offset / size
+            chunk = f.read(CHUNK_SIZE)
+            offset = len(chunk)
+
+            upload_session = dbx.files_upload_session_start(chunk)
+
+            while True:
+                chunk = f.read(CHUNK_SIZE)
+                if not chunk:
+                    break
+                helper.progress = offset / size
                 helper.save()
-                try:
-                    uploader.upload_chunked()
-                except ErrorResponse as e:
-                    if retries < max_retries:
-                        retries += 1
-                    else:
-                        helper.failure_reason = str(e)
-                        helper.save()
-                        raise e
+                dbx.files_upload_session_append_v2(
+                    chunk,
+                    UploadSessionCursor(
+                        upload_session.session_id,
+                        offset,
+                    ),
+                )
+                offset += len(chunk)
 
-            upload = uploader.finish(helper.dest)
+            file_metadata = dbx.files_upload_session_finish(
+                b'',
```

Comment:
in order to finish the upload you must pass some bytes in i believe. honestly though the documentation is pretty bad and the way i figured out how this worked was looking at the source code, so there could be a better way. i was able to sync a ccz to my dropbox, download it and unzip the ccz and it seemed fine